specs/_features/eip7594/p2p-interface.md

```diff
@@ -0,0 +1,180 @@
+# EIP-7594 -- Networking
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Modifications in EIP-7594](#modifications-in-eip-7594)
+  - [Preset](#preset)
+  - [Configuration](#configuration)
+  - [Containers](#containers)
+    - [`DataColumnSidecar`](#datacolumnsidecar)
+    - [`DataColumnIdentifier`](#datacolumnidentifier)
+  - [Helpers](#helpers)
+      - [`verify_data_column_sidecar_kzg_proof`](#verify_data_column_sidecar_kzg_proof)
+      - [`verify_data_column_sidecar_inclusion_proof`](#verify_data_column_sidecar_inclusion_proof)
+      - [`compute_subnet_for_data_column_sidecar`](#compute_subnet_for_data_column_sidecar)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Samples subnets](#samples-subnets)
+        - [`data_column_sidecar_{subnet_id}`](#data_column_sidecar_subnet_id)
+  - [The Req/Resp domain](#the-reqresp-domain)
+    - [Messages](#messages)
+      - [DataColumnSidecarByRoot v1](#datacolumnsidecarbyroot-v1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Modifications in EIP-7594
+
+### Preset
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH`   | `uint64(floorlog2(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments')))` (= 4) | <!-- predefined --> Merkle proof index for `blob_kzg_commitments` |
+
+### Configuration
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `DATA_COLUMN_SIDECAR_SUBNET_COUNT`              | `32`                               | The number of data column sidecar subnets used in the gossipsub protocol.  |
+
+### Containers
+
+#### `DataColumnSidecar`
+
+```python
+class DataColumnSidecar(Container):
+    index: ColumnIndex  # Index of column in extended matrix
+    column: DataColumn
+    kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    kzg_proofs: List[KZGProof, MAX_BLOB_COMMITMENTS_PER_BLOCK]
```

Comment:
The type of `DataColumn` is restricted to `MAX_BLOBS_PER_BLOCK`. Do we need a new type for this? Or maybe to change the max length for `DataColumn`. From `List[Cell, MAX_BLOBS_PER_BLOCK]` to `List[Cell, MAX_BLOB_COMMITMENTS_PER_BLOCK]`.

Reply:
yes I also agree we should modify the `DataColumn` to have `MAX_BLOB_COMMITMENTS_PER_BLOCK` max length for keeping it `fork` free

Reply:
updated in https://github.com/ethereum/consensus-specs/pull/3574/commits/bb33f9082c552af101a96ffbd9e973cb40464d7a

thanks!